Make gaussHermiteData callable from other packages.

kollerma opened the pull request:
Hi

I'd like to use your methods from within my package (robustlmm). This patch makes the functions available, so I can just call them from my package's c++ code.

Best,
Manuel

kollerma committed:
Make gaussHermiteData callable from other packages.

awblocker commented:
Thanks for doing this, and apologies for the delay on my end. I think this would need a corresponding update to gaussHermiteData in R/lib.R.

kollerma commented:
I assume you're talking about the rename of gaussHermiteData? There were two functions with that name and that didn't play well with the export part. So I've renamed one of them. The one called from R is still called gaussHermiteData. Increasing the version number should be the only thing you need to do for a new release.

kollerma committed:
Improving header file to auto load package if required.

This way fastGHQuad only needs to be added to the other
packages’ LinkingTo. No need to add it to Depends and
add dummy imports.

kollerma committed:
Fixing R CMD check NOTE about register symbols.

As suggested here:
https://github.com/RcppCore/Rcpp/issues/636#issuecomment-280985661
